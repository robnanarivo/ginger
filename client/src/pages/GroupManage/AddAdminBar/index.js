import AddAdminBar from './AddAdminBar';

export default AddAdminBar;
