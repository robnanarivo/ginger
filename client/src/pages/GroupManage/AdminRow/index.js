import AdminRow from './AdminRow';

export default AdminRow;
