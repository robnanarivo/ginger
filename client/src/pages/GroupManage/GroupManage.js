import React, { useState, useEffect } from 'react';
import ListSubheader from '@mui/material/ListSubheader';
import List from '@mui/material/List';
import Box from '@mui/material/Box';
import Divider from '@mui/material/Divider';
import { useParams } from 'react-router-dom';
import AdminRow from './AdminRow';
import AddAdminBar from './AddAdminBar';
import { getGroupData } from '../../api';

function displayAdmins(listOfAdmins, updateAdmins, creatorId) {
  const res = [];
  let count = 0;
  listOfAdmins.forEach((admin) => {
    count += 1;
    res.push(
      <AdminRow
        key={admin.userId}
        adminId={admin.id}
        adminPic={admin.profilePicture}
        adminName={admin.userName}
        updateAdmins={updateAdmins}
        creatorId={creatorId}
      />,
    );
    if (count !== listOfAdmins.length) {
      res.push(<Divider key={`d${count}`} />);
    }
  });
  return res;
}

export default function GroupManage() {
  const { groupId } = useParams();
  const [admins, setAdmins] = useState([]);
  const [groupData, setGroupData] = useState({ users: [], creatorId: '' });
  useEffect(async () => {
    const res = await getGroupData(groupId);
    setAdmins(res.admins);
    setGroupData(res);
  }, []);

  const updateAdmins = async () => {
    const res = await getGroupData(groupId);
    setAdmins(res.admins);
    setGroupData(res);
  };

  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'center',
        backgroundColor: '#F4F3F5',
        minHeight: '100vh',
      }}
    >
      <Box
        sx={{
          display: 'flex',
          flexDirection: 'column',
          width: '800px',
          marginTop: '50px',
        }}
      >
        <List
          sx={{
            width: '100%',
            marginTop: 8,
            maxWidth: 720,
            bgcolor: '#FFFFFF',
            alignSelf: 'flex-start',
            marginBottom: 8,
          }}
          component="nav"
          aria-labelledby="nested-list-subheader"
          subheader={
            (
              <ListSubheader component="div" id="nested-list-subheader" sx={{ bgcolor: '#E5E5E5' }}>
                Group Admin
              </ListSubheader>
            )
          }
        >
          {displayAdmins(admins, updateAdmins, groupData.creatorId)}
        </List>
        <AddAdminBar users={groupData.users} updateAdmins={updateAdmins} admins={admins} />
      </Box>
    </Box>
  );
}
