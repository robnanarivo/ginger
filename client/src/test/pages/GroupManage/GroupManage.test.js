import React from 'react';
import renderer from 'react-test-renderer';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
// import { render, screen } from '@testing-library/react';
import GroupManage from '../../../pages/GroupManage/GroupManage';
import 'jest-styled-components';

describe('AdminRow unit test', () => {
  it('AdminRow Snapshot test', () => {
    const history = createMemoryHistory();
    const tree = renderer
      .create(
        <Router history={history}>
          <GroupManage />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
