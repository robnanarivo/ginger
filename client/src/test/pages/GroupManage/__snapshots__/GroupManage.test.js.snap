// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminRow unit test AdminRow Snapshot test 1`] = `
<div
  className="MuiBox-root css-dbssi2"
>
  <div
    className="MuiBox-root css-1gttelj"
  >
    <nav
      aria-labelledby="nested-list-subheader"
      className="MuiList-root MuiList-padding MuiList-subheader css-jns7y-MuiList-root"
    >
      <div
        className="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky css-ap3lwe-MuiListSubheader-root"
        id="nested-list-subheader"
      >
        Group Admin
      </div>
    </nav>
    <div
      className="MuiBox-root css-11fmhea"
    >
      <div
        aria-expanded={false}
        aria-owns={null}
        className="MuiAutocomplete-root css-3ja3g8-MuiAutocomplete-root"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="combobox"
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="free-solo"
            id="free-solo-label"
          >
            User Name
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-controls={null}
              aria-invalid={false}
              autoCapitalize="none"
              autoComplete="off"
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputTypeSearch MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1j5luls-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="free-solo"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              required={false}
              spellCheck="false"
              type="search"
              value=""
            />
            <div
              className="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-173wfxe"
              >
                <span>
                  User Name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-dcq3k7-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        ADD
      </button>
    </div>
  </div>
</div>
`;
